### PR TITLE
docs: document balancing cores and max-jobs

### PR DESCRIPTION
**Pull request by grahamc**
Write a bit about the balance between max-jobs and cores.

### CODE REVIEW

File: doc/manual/advanced-topics/cores-vs-jobs.xml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+will be utilized: <xref linkend="conf-cores" /> and
+<xref linkend="conf-max-jobs" />.</para>
+
+<variablelist>
+  <varlistentry>
+    <term><xref linkend="conf-max-jobs" /></term>
+    <listitem><para>
+      Dictates how many separate derivations will be built at the same
+      time. If you set this to zero, the local machine will attempt to
+      do no builds (forcing it to be built remotely.)
```
Comment (edolstra) [inline doc/manual/advanced-topics/cores-vs-jobs.xml]: `will attempt to do no build` -> `will do no builds`.

File: doc/manual/advanced-topics/cores-vs-jobs.xml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    </para></listitem>
+  </varlistentry>
+  <varlistentry>
+    <term><xref linkend="conf-cores" /></term>
+    <listitem><para>
+      Dictates how many cores each derivation is allowed to use. If
+      you set this to zero, the builds will be given access to all the
+      cores of your system.
```
Comment (edolstra) [inline doc/manual/advanced-topics/cores-vs-jobs.xml]: "Dictates" -> "suggests" (and drop the second sentence). There is nothing that stops a derivation from using more cores. In practice, the "cores" option translates to GNU Make's `-j <cores>` flag.

File: doc/manual/advanced-topics/cores-vs-jobs.xml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      <td>
+        24 derivations can build at the same time, each using all the
+        available cores of the machine. Very likely to be oversold,
+        and very likely to suffer context switches.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+<para>It is up to the derivations' build scirpt to respect
```
Comment (edolstra) [inline doc/manual/advanced-topics/cores-vs-jobs.xml]: `scirpt` -> `script.

**Comment by jonringer** [comment]
ping @grahamc 

I still  think this is really valuable, and would love to have it in the manual :)

**Comment by grahamc** [comment]
Thanks @jonringer, I've updated the docs and addressed @edolstra's feedback. Would you might giving it a read for mistakes? I have a tendency to make silly mistakes :x

File: doc/manual/advanced-topics/advanced-topics.xml [hunk 4]
```diff
@@ -7,6 +7,8 @@
 <title>Advanced Topics</title>
 
 <xi:include href="distributed-builds.xml" />
+<xi:include href="cores-vs-jobs.xml" />
 <xi:include href="diff-hook.xml" />
 
+
```
Comment (jonringer) [inline doc/manual/advanced-topics/advanced-topics.xml]: Extra white space? I don't see 2 newlines anywhere else.
```suggestion
```

Comment (grahamc) [reply doc/manual/advanced-topics/advanced-topics.xml]: Thanks! Fixed up!

**Pull request merged**